beta-src/src/components/ui/WDMain.tsx

```diff
@@ -253,6 +258,78 @@ const WDMain: React.FC = function (): React.ReactElement {
       isLivePhase: false,
     };
   };
+
+  useEffect(() => {
+    Object.entries(ordersMeta).forEach(([_, orderMeta]) => {
+      // Let's use as an example ION C APU - TUN
+      // The board should automatically also enters APU - TUN (Via Convoy)
+      // More info in CHR-68
+      const orderMetaUpdate = orderMeta.update;
+
+      if (
+        orderMetaUpdate?.type === "Convoy" &&
+        orderMetaUpdate?.fromTerrID &&
+        orderMetaUpdate.toTerrID
+      ) {
+        const fromTerr = maps.terrIDToProvince[orderMetaUpdate.fromTerrID];
+        const toTerr = maps.terrIDToProvince[orderMetaUpdate.toTerrID];
+
+        const legalConvoysByUnitIDs = Object.keys(
+          legalOrders.hasAnyLegalConvoysByUnitID,
+        );
+        if (legalConvoysByUnitIDs.length === 1) {
```

Comment:
Why does this need to be length 1? Is it intended for this code not to work if there is more than one unit that has a possible convoy, or am I misunderstanding what the control logic is here?

Reply:
Also, are we making assumptions about what keys hasAnyLegalConvoysByUnitID is supposed to have? It's a map from unitID -> boolean, so it may possibly have entries for units that don't have a legal convoy, and just have a value of "false" for those units.

Reply:
Yes, it needs to be 1 in case there is another unit of the same country with a valid convoy. If there is only one, the "Via convoy" is automatically applied

Reply:
actually, your observation was good, I tried with another unit in the territory where the convoy would suppose to go and it would cause a conflict. I just removed that validation.